ChurchCRM/CRM
# Added logic to 'create' INSERT sql to avoid SQL strict mode errors.

#### What's this PR do?
Ported fix by @chiebert to handle Event Type creation when MySQL strict mode is enabled (which seems more common than we first expected.

#### Screenshots (if appropriate)
N/A

#### What Issues does it Close?

#4273 #4290 

#### What are the relevant tickets?

#4273 #4290 

#### Any background context you want to provide?

There is a lot of relevant discussion on #4273 which outlines the root cause and potential fixes. Rather than expecting people to modify their MySQL setups (which for some shared hosting setups is impossible) we have opted to handle this in our code instead.

#### Where should the reviewer start?

#4273 - read the comments and discussion. Then review this PR.

#### How should this be manually tested?

1. Ensure a problematic strict mode is enabled in your MySQL setup (see [MySQL Documentation](https://dev.mysql.com/doc/refman/en/sql-mode.html#sql-mode-strict)) which should return either `STRICT_TRANS_TABLES` or `STRICT_ALL_TABLES`, eg.
```
mysql> SELECT @@GLOBAL.sql_mode;
+-------------------------------------------------------------------------------------------------------------------------------------------+
| @@GLOBAL.sql_mode                                                                                                                         |
+-------------------------------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+-------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT @@SESSION.sql_mode;
+-------------------------------------------------------------------------------------------------------------------------------------------+
| @@SESSION.sql_mode                                                                                                                        |
+-------------------------------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+-------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
2. Go to http://192.168.33.10/EventNames.php and create a new event type.
3. New event type should now display in the list.

#### How should the automated tests treat this?
Creating an event type should success regardless of database configuration.

#### Questions:
- Is there a related website / article to substantiate / explain this change?
  -  [x] Yes - https://dev.mysql.com/doc/refman/en/sql-mode.html#sql-mode-strict

- Does the development wiki need an update?

- Does the user documentation wiki need an update?

- Does this add new dependencies?

- Does this need to add new data to the demo database
